_source/_posts/2020-04-22-build-secure-aws-lambda-nodejs-react.md

```diff
@@ -0,0 +1,1057 @@
+
+Serverless architecture with AWS Lambdas is quickly becoming a popular option for companies looking to deploy applications without the overhead of maintaining servers. AWS Lambda functions are event-driven and serverless—triggered to process a piece of code and return a result. AWS Lambdas can be written in most common languages today on a variety of platforms including .NET Core, Java, Go and, in the case of this post, Node.js.
+
+The example in this post uses Node.js 12.x, the latest version available on AWS to date. You'll be using AWS APIs to access AWS Lambdas from your application. This post will teach you how to write your API documentation in YAML and import it into AWS APIs. Additionally, you'll learn how to deploy that API and enable CORS so that your application can access it.
+
+For this project, you'll be using React.js for the front end. React.js is one of the most popular javascript front end frameworks. React makes creating interactive UIs simple and intuitive. To authenticate users, you'll be using Okta's Single Sign-on Provider as well as Okta's React libraries, which make handling authentication very simple, allowing you more time to focus on your business code.
+
+This post outlines how to make a simple version of one of my favorite game shows as a kid, Card Sharks. The premise of the game is simple enough: A player is shown six cards, one face-up and five face-down. The player must guess if the first face-down card is higher or lower than the current face-up card. If the player is correct they gain a point and move to the next card. If the card has the same rank, no point is awarded and the game continues. If the player is incorrect, the game is over. If the player gets all the cards correct they are allowed to reshuffle the deck and draw a fresh set of 6 cards. At the end of the game, the player can submit their high score.
+
+## Get Started with Okta
+
+First thing's first, set up your [Okta application](https://developer.okta.com). Log into your Okta admin portal and navigation to *Applications*. Click the button labeled *Add Application*, select *Single-Page App,* then click *Next*. Name your application something that will represent its role for you. In my case, I used "Card Sharks."  If your application uses port 8080, you can continue to use `http://localhost:8080` for your URLs. However, if you aren't planning on specifically designating port 8080 for your application you will most likely use port 3000 for the application and will need to change these URLs to reflect that.
+
+{% img blog/build-secure-aws-lambda-nodejs-react/app-settings.jpg alt:"Okta Application Settings" width:"800" %}
+
+Your completed Okta set up should look similar to this.
+
+Click *Done* to review your application setup. Make sure to grab the *Client ID* listed under *Client Credentials* as it will be needed in your react.js application.
+
+## Setup AWS
+
+Next, set up AWS to handle your serverless operations.
+
+### Set up Lambda Functions
```

Comment:
Should be consistent on how we spell setup - one word or two. Also, capitalization of Up if two.